File: nodejs/checkio/src/home/split_list.ts
import assert from "assert";

function splitList<T>(values: T[]): [T[], T[]] {
	const splitAt = Math.ceil(values.length / 2);
	return [values.slice(0, splitAt), values.slice(splitAt)];
}

console.log("Example:");
console.log(JSON.stringify(splitList([1, 2, 3, 4, 5, 6])));

// These "asserts" are used for self-checking
assert.deepStrictEqual(splitList([1, 2, 3, 4, 5, 6]), [
	[1, 2, 3],
	[4, 5, 6],
]);
assert.deepStrictEqual(splitList([1, 2, 3]), [[1, 2], [3]]);
assert.deepStrictEqual(
	splitList(["banana", "apple", "orange", "cherry", "watermelon"]),
	[
		["banana", "apple", "orange"],
		["cherry", "watermelon"],
	],
);
assert.deepStrictEqual(splitList([1]), [[1], []]);
assert.deepStrictEqual(splitList([]), [[], []]);

console.log("Coding complete? Click 'Check Solution' to earn rewards!");

File: python/checkio/home/split-list.py
from typing import Any, Iterable


def split_list(items: list[Any]) -> Iterable[list[Any]]:
    if len(items) % 2 == 0:
        return [items[: len(items) // 2], items[len(items) // 2 :]]
    else:
        return [items[: len(items) // 2 + 1], items[len(items) // 2 + 1 :]]


print("Example:")
print(list(split_list([1, 2, 3, 4, 5, 6])))

assert list(split_list([1, 2, 3, 4, 5, 6])) == [[1, 2, 3], [4, 5, 6]]
assert list(split_list([1, 2, 3])) == [[1, 2], [3]]
assert list(split_list(["banana", "apple", "orange", "cherry", "watermelon"])) == [
    ["banana", "apple", "orange"],
    ["cherry", "watermelon"],
]
assert list(split_list([1])) == [[1], []]
assert list(split_list([])) == [[], []]

print("The mission is done! Click 'Check Solution' to earn rewards!")
